docs/guides/high-availability/concepts-threads.adoc

```diff
@@ -0,0 +1,69 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Concepts for configuring thread pools"
+summary="Understand these concepts to avoid resource exhaustion and congestion"
+preview="true"
+tileVisible="false" >
+
+
+This section is intended when you want to understand the considerations and best practices on how to configure thread pools connection pools for {project_name}.
+For a configuration where this is applied, visit <@links.ha id="deploy-keycloak-kubernetes" />.
+
+== Concepts
+
+=== Quarkus executor pool
+
+{project_name} requests, as well as all probes, are handled by the Quarkus executor pool.
+
+The Quarkus executor thread pool is configured in https://quarkus.io/guides/all-config#quarkus-core_quarkus.thread-pool.max-threads[`quarkus.thread-pool.max-threads`] and has a maximum size of at least 200 threads.
+Depending on the available CPU cores, it can grow even larger.
+Threads are created as needed, and will end when no longer needed, so the system will scale up and down as needed.
+
+When the load and the number of threads increases, the bottleneck will usually be the database connections.
```

Comment:
I would also say CPU because you will have more context switches/thread contention. Basically, more threads do not necessarily mean more processing power but it can be the opposite as you are pushing too much on the CPU. If you have enough cores available, that is fine. Otherwise, you should consider lowering the number of threads in favor of faster response times (less context switches/contention) and better throughput.

In addition to what is being said here I would also consider:

```
You should also consider adjusting the number of worker threads to avoid creating more than what the CPU limits set to nodes can handle. Ideally, you should start with a low limit of threads and adjust it accordingly to the target throughput and response time.
```